core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -616,6 +621,7 @@ public static String typeName(byte type) {
     }
 
     private static final class Fields {
+        static final String TYPE = new String("type");
```

Comment:
I'd remove the whole `Fields` class. They have fallen out of favor. I'd just use the string at the call site, especially if it is just used once.